### PR TITLE
Add JIT Hooks Documentation

### PR DESCRIPTION
**Pull request by dsouzai**

### CODE REVIEW

File: doc/compiler/runtime/JITHooks.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+|`J9HOOK_VM_INITIALIZE_SEND_TARGET`|Notifies when J9Methods are to have their send target initialized.|Used to set initial invocation counts.|
+|`J9HOOK_MM_OMR_LOCAL_GC_START`|Notifies when a local Garbage Collection has started.|Local here applies to policies such as GenCon, where only the nursery region is being GC'd.|
+|`J9HOOK_MM_OMR_LOCAL_GC_END`|Notifies when a local GC has ended|If Real Time GC is enabled, the JIT registers `jitHookLocalGCEnd`; otherwise it registers `jitHookReleaseCodeLocalGCEnd`.|
+|`J9HOOK_MM_OMR_GLOBAL_GC_START`|Notifies when a global GC has started.|Global here implies a GC of the entire heap.|
+|`J9HOOK_MM_OMR_GLOBAL_GC_END`|Notifies when a global GC has ended.|If Real Time GC is enabled, the JIT registers `jitHookGlobalGCEnd`; otherwise it registers `jitHookReleaseCodeGlobalGCEnd`.|
+|`J9HOOK_MM_OMR_GC_CYCLE_END`|Notifies when the GC Cycle has ended.|The JIT only registers against this hook if Real Time GC is enabled.|
+|`J9HOOK_VM_INTERNAL_CLASS_LOAD`|Notifies when a class has been loaded.||
+|`J9HOOK_VM_CLASS_PREINITIALIZE`|Notifies prior to notify the initialization of a class prior to running the `<clinit>`.|
```
Comment (mpirvu) [inline doc/compiler/runtime/JITHooks.md:45]: This phrasing is a bit cumbersome. Could we make it clearer with something like: `Notifies prior to initialization of a class through the <clinit> method`

Comment (dsouzai) [reply doc/compiler/runtime/JITHooks.md:45]: Ah good catch, the phrasing was different before, and I just did a mass search/replace which clearly didn't work out in this case.

Comment (dsouzai) [reply doc/compiler/runtime/JITHooks.md]: Fixed in https://github.com/eclipse/openj9/pull/10205/commits/42c8adf0e62227dce6137f78649803fb61249ef1

**Comment by mpirvu** [comment]
@dsouzai Do you plan to to make changes? If not, this PR could be merged.

**Comment by dsouzai** [comment]
> @dsouzai Do you plan to to make changes? If not, this PR could be merged.

Nope, I figure this should be good enough for a first iteration; if there's more to be added it can be done in a future PR.

**Pull request merged**